ZupIT/ritchie-cli
# [Feature] Adding ruby language support

**- What I did**
Added ruby language support for when creating a new formula according to issue #186 . **I have intentionally left out the coffee scaffold as I think it will be discontinued**, if not, please let me know

To provide more coverage on fail cases, I had to resort to dependency injection on `createSrcFiles` by extracting an interface on `CreateManager`. Ended up running this fail case test to all other languages and raised the coverage on `createSrcFiles` from 75% to 92.9%

Removed an extraneous param used on the languages `New` function

Fixed a small bug in which failed `Shell` language creation would return `nil` and not account for the error

**- How to verify it**
you can run `rit create formula` and then select the `ruby` language. The appropriate `rb` files along with the `Gemfile` are created. You can also verify it through `make unit-test`

<img width="555" alt="Screen Shot 2020-07-07 at 15 52 11" src="https://user-images.githubusercontent.com/3487122/86866318-16094880-c0a7-11ea-8106-e1680182257d.png">

**- Description for the changelog**
* Added support to write new formulas in ruby
